gen/calc.py

```diff
@@ -303,83 +304,122 @@ def validate_network_default_name(overlay_network_default_name, dcos_overlay_net
             overlay_network_default_name))
 
 
-def validate_dcos_ucr_default_bridge_subnet(dcos_ucr_default_bridge_subnet):
+class IPVersion(IntEnum):
+    """Defines the package source to install package from
+    """
+    IPv4 = 4
+    IPv6 = 6
+
+
+def validate_config_subnet(config_name, subnet, version=IPVersion.IPv4):
     try:
-        ipaddress.ip_network(dcos_ucr_default_bridge_subnet)
+        network = ipaddress.ip_network(subnet)
+        if (version == IPVersion.IPv4 and not isinstance(network, ipaddress.IPv4Network)) or \
+                (version == IPVersion.IPv6 and not isinstance(network, ipaddress.IPv6Network)):
+            raise ValueError("IP version not match")
     except ValueError as ex:
-        raise AssertionError(
-            "Incorrect value for dcos_ucr_default_bridge_subnet: {}."
-            " Only IPv4 subnets are allowed".format(dcos_ucr_default_bridge_subnet)) from ex
+        err_msg = "Incorrect value for {}: {}. Only IPv{} subnets are allowed".format(
+            config_name, subnet, version)
+        raise AssertionError(err_msg) from ex
 
 
 def validate_dcos_overlay_network(dcos_overlay_network):
+    # a validate dcos_overlay_network is in such a format:
```

Comment:
What do you think about using here jsonschema instead of if/then/else stanzas?

Reply:
Thanks for this comment. But I am afraid jsonschema can only validate only parts of the format, and to keep a minimal change, I suggest we don't introduce jsonschema here.